planner/core/expression_rewriter.go

```diff
@@ -1267,5 +1267,12 @@ func (er *expressionRewriter) toColumn(v *ast.ColumnName) {
 			return
 		}
 	}
+	if _, ok := er.p.(*LogicalUnionAll); ok && v.Table.O != "" {
+		er.err = ErrTablenameNotAllowedHere.GenWithStackByArgs(v.Table.O, "SELECT", clauseMsg[er.b.curClause])
+		return
+	}
+	if er.b.curClause == globalOrderByClause {
+		er.b.curClause = orderByClause
```

Comment:
Why we edit it here?

Reply:
To produce the same error like MySQL when the column name is not valid:

https://github.com/pingcap/tidb/blob/322795439dafb0bc6c17cebd598a0520926db94b/executor/prepared_test.go#L335-L351